### ISSUE TITLE
Make Cvode solver work with Sundials 2 and 3

### ISSUE THREAD

**Issue by MichaelClerx** [ModellingWebLab/weblab-fc#7]
Sundials 2 is no longer packaged for fedora, so am unable to test `fc` on my machine

Sundials version number is available from e.g.:

```
#include <sundials/sundials_config.h>
#ifdef SUNDIALS_PACKAGE_VERSION
    return PyUnicode_FromString(SUNDIALS_PACKAGE_VERSION);
#else
    return PyUnicode_FromString(SUNDIALS_VERSION);
#endif
```

so shouldn't be too hard to write code for both?

**Comment by jonc125** [comment]
https://stackoverflow.com/questions/26225187/try-statement-in-cython-for-cimport-for-use-with-mpi4py looks like a potential way forward

**Comment by jonc125** [comment]
See https://chaste.cs.ox.ac.uk/trac/changeset/04c367eed610358fe42f8ba97355ada2b84b131b/git_repo for how Chaste supports both versions.

**Comment by MichaelClerx** [comment]
https://grokbase.com/t/gg/cython-users/128rn1399p/equivalent-of-ifdef-preprocessor

**Comment by MichaelClerx** [comment]
```
import cython

code = '''
cdef extern from *:
    """
    #include <sundials/sundials_config.h>

    #ifndef SUNDIALS_VERSION_MAJOR
        #define SUNDIALS_VERSION_MAJOR 2
    #endif
    """
    int SUNDIALS_VERSION_MAJOR

return SUNDIALS_VERSION_MAJOR
'''

print(cython.inline(code))
```

**Comment by MichaelClerx** [comment]
Closed in #10
